tools/run_tests/artifacts/build_artifact_php.sh

```diff
@@ -21,12 +21,5 @@ mkdir -p "${ARTIFACTS_OUT}"
 
 # Build the PHP extension archive (this just zips all the files up)
 pear package
-# Note: the extension compiled by this step is not being used in any
-# way, i.e. they are not the pacakge being distributed.
-# This is done here to get an early signal for compiling the PHP
-# extension in some form.
-find . -name "grpc-*.tgz" | cut -b3- | xargs pecl install
```

Comment:
So this step (trying to install the pecl package) is basically an early distribtests and it's the same thing as what's being done by all the PHP distribtests:
https://github.com/grpc/grpc/blob/013a45ccc4f0cc99532cb3148ca6ed3f60996711/test/distrib/php/run_distrib_test.sh#L22

So from that perspective the step in build_artifacts jobs is duplicate and it also strictly speaking doesn't belong there. The reason why we were including this step until now is that we didn't have a simple way to run PHP distribtests on PRs, so this comand was added just be able to detect potential distribtest breakages early.

This PR solves this problem by making it possible to run all PHP distribtests on PR. Trying to install the pear package as part of artifact build can be removed since it doesn't add any new information and only slows down the artifact build (which is very fast otherwise, since it basically just created a tarball with all the files).

Reply:
Thanks for clarifying. Yes that brings back my memory of why we are doing the `pecl install` step in the current artifact PR test. 

Yes the whole point is that we definitely want to run the `pecl install` step in PR tests. Previously it's only possible in the artifact test. Now if we can run the distribtests in PR tests too, then yes we only need to do it there.